collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2432,7 +2419,7 @@ public void denied() {
                             savedFormStart = true;
                             formController.getAuditEventLogger().logEvent(AuditEvent.AuditEventType.FORM_RESUME, null, true);
                             formController.getAuditEventLogger().logEvent(AuditEvent.AuditEventType.HIERARCHY, null, true);
-                            startActivity(new Intent(this, FormHierarchyActivity.class));
+                            startActivityForResult(new Intent(this, FormHierarchyActivity.class), RequestCodes.HIERARCHY_ACTIVITY);
```

Comment:
This makes the back button work as intended by going to `onActivityResult` where we can rebuild the view.